lib/semantics/scope.cc

```diff
@@ -25,6 +25,29 @@ namespace Fortran::semantics {
 
 Symbols<1024> Scope::allSymbols;
 
+bool EquivalenceObject::operator==(const EquivalenceObject &that) const {
+  return symbol == that.symbol && subscripts == that.subscripts;
+}
+
+bool EquivalenceObject::operator<(const EquivalenceObject &that) const {
+  return &symbol < &that.symbol ||
+      (&symbol == &that.symbol && subscripts < that.subscripts);
+}
+
+std::string EquivalenceObject::AsFortran() const {
+  std::stringstream ss;
+  ss << symbol.name().ToString();
+  if (!subscripts.empty()) {
+    char sep{'('};
+    for (auto subscript : subscripts) {
+      ss << sep << subscript;
+      sep = ',';
```

Comment:
Should the separator be `:` for characters ?
Currently, `-fdebug-dump-symbols` dump of  is `equivalence (c(5:10), d)` where `c` and `d` are scalar characters is `Equivalence Sets: (c(5,10),d)` .